source/taskmanager/src/main/java/nl/aerius/taskmanager/mq/RabbitMQMessageHandler.java

```diff
@@ -100,9 +100,7 @@ public void onMessageReceived(final RabbitMQMessage message) {
 
   @Override
   public void onConsumerShutdown(final ShutdownSignalException sig) {
-    if (!sig.isInitiatedByApplication()) {
-      tryStartConsuming();
-    }
+    tryStartConsuming();
```

Comment:
This won't cause issues when consumer isn't null in `stopAndStartConsumer`?

The `tryStartConsuming` will use that method to stop the existing consumer and start a new one, and stopping the existing will cause a `ShutdownSignalException`, which will end up in here again, which will try `tryStartConsuming` again, which....  Not sure if that's the case, but perhaps the consumer should be set to `null` in this method if it is?

Reply:
To avoid this case I've added an AtomicBoolean that is set during trying-to-connect. wdyt?

Reply:
Yeah, think that works